src/beagle-view/render/index.ts

```diff
@@ -161,21 +163,69 @@ function createRenderer({
 
     renderToScreen(view)
   }
-  
+
   function doFullRender(
     viewTree: BeagleUIElement<any>,
     anchor = '',
     mode: TreeUpdateMode = 'replaceComponent',
   ) {
     viewTree = runLifecycle(viewTree, 'beforeStart')
+
     let viewTreeWithIds = preProcess(viewTree)
     viewTreeWithIds = runLifecycle(viewTreeWithIds, 'beforeViewSnapshot')
+
     doPartialRender(viewTreeWithIds, anchor, mode)
   }
 
+  function doTemplateRender(
+    templateManager: TemplateManager,
+    anchor: string,
+    contexts: DataContext[][],
+    componentManager?: ComponentManager
+  ) {
+    if (!templateManager.default && (!templateManager.templates || templateManager.templates.length === 0)) {
+      throw new BeagleParseError('The template to render was not provided!')
+    }
+    if (!anchor) throw new BeagleParseError('The anchor id to render a template was not provided!')
+    if (!contexts || contexts.length === 0) throw new BeagleParseError('At least one data context should be provided!')
+
+    const uiTree = beagleView.getTree()
+    const anchorElement = Tree.findById(uiTree, anchor)
+
+    if (!anchorElement) throw new BeagleParseError(`The anchor element with id "${anchor}" was not found!`)
+
+    const getTreeContextHierarchy = (uiTree: IdentifiableBeagleUIElement, globalContexts: DataContext[]) => {
+      const hierarchy = Context.evaluate(uiTree, globalContexts, false)
+      return Object.keys(hierarchy).map(key => hierarchy[key]).reduce((prev, cur) => [...prev, ...cur], [])
+    }
+
+    const globalContexts = [beagleView.getBeagleService().globalContext.getAsDataContext()]
+    const treeContextHierarchy = getTreeContextHierarchy(uiTree, globalContexts) || []
+    let shouldRender = false
+
+    contexts.forEach((context, index) => {
+      const contextHierarchy = [...treeContextHierarchy, ...context || []]
```

Comment:
the order matters here. The lower the index, the closest it is in the context hierarchy.
```suggestion
      const contextHierarchy = [...context || [], ...treeContextHierarchy]
```